### DISCUSSION TITLE
Error "An unexpected 'PrimitiveValue' node was found when reading from the JSON reader." on SK

### DISCUSSION THREAD

**Discussion by angeldelolmo** [microsoft/semantic-kernel#3293] [Q&A] [answered]
Hi everyone:

We're developing a copilot platform with Semantic Kernel, Azure OpenAI and Cognitive Search.
The first step is indexing documentation in Cognitive Search vector index.

This is a summary of the code to create embedding and store them in ACS:

`
    kernel = sk.Kernel()

    kernel.add_text_embedding_generation_service("ada", 
                                                AzureTextEmbedding("text-embedding-ada-002", 
                                                                   os.environ["AZURE_OPENAI_ENDPOINT"], 
                                                                   os.environ["AZURE_OPENAI_API_KEY"]))

    connector = AzureCognitiveSearchMemoryStore(
        1536,
        os.environ["AZURE_COGNITIVE_SEARCH_ENDPOINT"], 
        os.environ["AZURE_COGNITIVE_SEARCH_ADMIN_KEY"]
    )

    kernel.register_memory_store(memory_store=connector)

    await kernel.memory.save_information_async("vectorIndex", id="info1", text="My name is Andrea")
`

And these are the columns in the vector index:

![image](https://github.com/microsoft/semantic-kernel/assets/80886772/1025f534-7b85-4bb5-93d6-83846b13af02)

We're receiving the error "An unexpected 'PrimitiveValue' node was found when reading from the JSON reader. A 'StartArray' node was expected." when the last line of the code is executed.

We're following some SK samples/demos in GitHub, but we dont find the schema we must create is ACS.

Any suggestions?

Thanks in Advanced.

**Comment by angeldelolmo** [comment]
For those who are interested:

We've resolved the issue creating the following columns in the ACS index:

![image](https://github.com/microsoft/semantic-kernel/assets/80886772/be2e813c-282d-416d-9404-fd2a982f2659)

But now we're facing another error when we execute a query:

`result = await kernel.memory.search_async("vectorIndex", question)`

The exception says:

Parameter name: $select
Code: Message: Invalid expression: Could not find a property named 'Id' on type 'search.document'.

Any suggestions?

Thanks in Advanced.

**Reply by KarthikS0801** [reply]
Hey can you explain how you fix this "An unexpected 'PrimitiveValue' node was found when reading from the JSON reader. A 'StartArray' node was expected." when the last line of the code is executed. error.
TIA.

**Comment by angeldelolmo** [comment]
We've reproduced the error executing the following query in ACS:

search=*&$select=Id

but the following query is working

search=*&$select=id

Means that SK is trying to retrieve the fileId "Id" (capital I) from ACS?
Is this a bug?

Thanks.

**Comment by angeldelolmo** [comment, marked as answer]
Hi:

We've recreated the ACS index changing the name of Id column (capital I) and now it works!!

Thanks.

**Comment by angeldelolmo** [comment]
Hi:

The code is working, but we're still receiving some errors:

Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x000002269B125360>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x000002269B1054E0>, 255610.765)]

SK is not closing connections after http requests?

Thanks.
